cloudify_cli/tests/test_execution_events_fetcher.py

```diff
@@ -36,78 +39,178 @@ def _mock_get(self, execution_id, from_event=0,
+        except EventProcessingTimeoutError:
+            pass
+        except Exception as e:
+            self.fail('Wrong exception raised; expected "{0}" but got "{1}"'
+                      .format('EventProcessingTimeoutError', type(e).__name__))
+
+    def test_events_processing_progress(self):
+        events_bulk1 = range(0, 5)
+        self.events = events_bulk1
+        events_fetcher = ExecutionEventsFetcher(self.client,
+                                                'execution_id',
+                                                batch_size=100)
+        import datetime
+        print '**** start time: ', datetime.datetime.now()
+        events_count = events_fetcher.fetch_and_process_events()
+        self.assertEqual(len(events_bulk1), events_count)
+        print '**** time after bulk1: ', datetime.datetime.now()
+        events_bulk2 = range(0, 10)
+        self.events.extend(events_bulk2)
+        events_count = events_fetcher.fetch_and_process_events()
+        self.assertEqual(len(events_bulk2), events_count)
+        print '**** time after bulk2: ', datetime.datetime.now()
+        events_bulk3 = range(0, 7)
+        self.events.extend(events_bulk3)
+        events_count = events_fetcher.fetch_and_process_events()
+        self.assertEqual(len(events_bulk3), events_count)
+        print '**** time after bulk3: ', datetime.datetime.now()
+
+    def test_wait_for_execution_invalid_timeout(self):
+        mock_execution = self.client.executions.get('deployment_id')
         try:
-            execution_events.fetch_events(get_remaining_events=True)
-            self.fail()
-        except RuntimeError:
+            wait_for_execution(self.client, 'deployment_id', mock_execution,
+                               timeout=1)
+            self.fail('Exception not raised; wait_for_execution was expected '
+                      'to raise ValueError but completed.')
+        except ValueError:
             pass
+        except Exception as e:
+            self.fail('Wrong exception raised; expected "{0}" but got "{1}"'
+                      .format('ValueError', type(e).__name__))
 
-    def test_events_progress(self):
+    def test_wait_for_execution_timeout(self):
         self.events = range(0, 5)
-        execution_events = ExecutionEventsFetcher(self.client,
-                                                  'execution_id',
-                                                  batch_size=100)
-        events = execution_events.fetch_events()
-        self.assertEqual(5, len(events))
-        self.events.extend(range(0, 10))
-        events = execution_events.fetch_events()
-        self.assertEqual(10, len(events))
-        self.events.extend(range(0, 5))
-        events = execution_events.fetch_events()
-        self.assertEqual(5, len(events))
+        mock_execution = self.client.executions.get('deployment_id')
+        try:
+            wait_for_execution(self.client, 'deployment_id', mock_execution,
```

Comment:
why not self.assertRaises?
Also, consider using timeouts as short as possible. These are unit tests and we should aim for the to run as quickly as possible. sometimes, this is hard to avoid, but it feels like it could be handled in this case